### ISSUE TITLE
Old `npm` engine version blocking installs with NPM 5/6

### ISSUE THREAD

**Issue by codeburke** [Blizzard/node-rdkafka#523]
**Environment Information**
 - OS [e.g. Mac, Arch, Windows 10]: **Mac High Sierra**
 - Node Version [e.g. 8.2.1]: **8.12.0**
 - NPM Version [e.g. 5.4.2]: **6.4.1**
 - C++ Toolchain [e.g. Visual Studio, llvm, g++]: **g++**
 - node-rdkafka version [e.g. 2.3.3]: **latest**

**Steps to reproduce**
1. Install Node 8.12.0+ and the associated NPM (6.4.1+)
2. Go to a project using `node-rdkafka`
3. `rm -rf node_modules && rm package-lock.json`
4. `npm install`

You will get an error complaining that `node-rdkafka` requires npm ^2.7.3 but the latest npm is 6.4+
```
npm ERR! notsup Unsupported engine for node-rdkafka@1.0.6: wanted: {"npm":"^2.7.3"} (current: {"node":"8.12.0","npm":"6.4.1"})
npm ERR! notsup Not compatible with your version of node/npm: node-rdkafka@1.0.6
npm ERR! notsup Not compatible with your version of node/npm: node-rdkafka@1.0.6
npm ERR! notsup Required: {"npm":"^2.7.3"}
npm ERR! notsup Actual:   {"npm":"6.4.1","node":"8.12.0"}
```

**node-rdkafka Configuration Settings**
N/A

**Additional context**
From looking at the Blame of package.json the npm version has never been changed. I don't know the exact version of NPM that started strictly enforcing the `engines` semver, but this is blocking teams from re-generating their package-lock.json (for good semver hygiene).

My suggestion would be to remove the `npm` engine dependency entirely and instead add a `node` engine dependency since `node-rdkafka` depends much more on the Node version than the NPM version.

**Comment by VolodyaG** [comment]
+1 
I tried `npm install node-rdkafka --save` but got `Unsupported engine for node-rdkafka@2.4.2: wanted: {"npm":"^2.7.3"} (current: {"node":"8.9.3","npm":"5.5.1"})`

**Comment by codeburke** [comment]
Thanks @VolodyaG I changed the title to indicate that NPM 5 users are also affected.

**Comment by webmakersteve** [comment]
Fixed with #524
